### PR TITLE
Arrays Intro: Fix grammar issues and use TextChatService

### PR DESCRIPTION
**Pull request by Flikter**
## Changes

Fixes grammar issues, adds spaces where needed, shortens code, uses TextChatService instead of deprecated Chat for dialogues, and fixes other minor issues.

## Checks

By submitting your pull request for review, you agree to the following:

- [x] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [x] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [x] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Comment by aetter** [comment]
Hey @Flikter, sorry for the late review here, but I've been having an extraordinarily hard time getting this to work properly in Studio unless I change the script `RunContext` to `Client` _while the experience is running_. Anything obvious I'm missing? Also flagging @othomson-roblox, since I can only assume this is a blind spot on my end.

**Comment by Flikter** [comment]
> Hey @Flikter, sorry for the late review here, but I've been having an extraordinarily hard time getting this to work properly in Studio unless I change the script `RunContext` to `Client` _while the experience is running_. Anything obvious I'm missing? Also flagging @othomson-roblox, since I can only assume this is a blind spot on my end.

Hello! No problem, I've just noticed that a variable was incorrectly incremented by 1, I've committed a fix for this. Thanks!

**Comment by aetter** [comment]
Hey @Flikter, I've looked at this for... let's call it more time than I'd like to admit. But I believe I've finally gotten to the bottom of it. 😄

It turns out that `TextChatService:DisplayBubble()` needs to run as a local script and (realistically) be parented to the starter player scripts. So the majority of the page would need a rewrite for `TextChatService` to work properly, because the script can't be parented to the proximity prompt.

I'm happy to (eventually) do the rewrite, but that means reverting much of this PR and just taking the `workspace`, whitespace, and wording, and other code improvements until that time. So I'm going to run through and change everything back to `Chat` and then merge. 👍

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ These steps use an array to store different phrases for the NPC to say when play
    ```lua
       -- Cycles through chat dialogue when prompt is used
 
-      local Chat = game:GetService("Chat")
+      local TextChatService = game:GetService("TextChatService")
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:51]:
```suggestion
      local Chat = game:GetService("Chat")
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 2]
```diff
@@ -57,11 +57,11 @@ These steps use an array to store different phrases for the NPC to say when play
       local head = characterParts.Head
 
       -- Add array here
-      local dialogueArray= {}
+      local dialogueArray = {}
 
       local function speak()
          local dialogue = "I've got one thing to say!"
-         Chat:Chat(head, dialogue)
+         TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:64]:
```suggestion
         Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 3]
```diff
@@ -111,11 +111,11 @@ Use index values to assign specific pieces of dialogue to the NPC. To use a valu
 1. Replace the variable `dialogue`'s default string value with index 2.
 
    ```lua
-    local dialogueArray= {"Hi!", "Do I know you?", "Goodbye!"}
+    local dialogueArray = {"Hi!", "Do I know you?", "Goodbye!"}
 
     local function speak()
       local dialogue = dialogueArray[2]
-      Chat:Chat(head, dialogue)
+      TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:118]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 4]
```diff
@@ -145,7 +145,7 @@ Whenever a player interacts with the NPC, increment the variable value by 1 to d
    ```lua
     local function speak()
       local dialogue = dialogueArray[dialogueIndex]
-      Chat:Chat(head, dialogue)
+      TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:148]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 5]
```diff
@@ -154,9 +154,9 @@ Whenever a player interacts with the NPC, increment the variable value by 1 to d
...
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:157]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 6]
```diff
@@ -185,27 +185,27 @@ Use the array size to check when it's time to cycle back to the first piece of d
...
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:188]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 7]
```diff
@@ -185,27 +185,27 @@ Use the array size to check when it's time to cycle back to the first piece of d
...
    ```
 
-2. If `dialogueIndex` isn't at the end, it should still add 1 to `dialogueIndex`. Move `dialogueIndex = dialogueIndex + 1` under an else statement.
+2. If `dialogueIndex` isn't at the end, it should still add 1 to `dialogueIndex`. Move `dialogueIndex += 1` under an else statement.
 
    ```lua
     local function speak()
       local dialogue = dialogueArray[dialogueIndex]
-      Chat:Chat(head, dialogue)
+      TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:203]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 8]
```diff
@@ -223,7 +223,7 @@ This script used an array to create a list of possible dialogue lines for a Non-
 ```lua title="Completed script"
    -- Cycles through chat dialogue when prompt is used
 
-   local Chat = game:GetService("Chat")
+   local TextChatService = game:GetService("TextChatService")
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:226]:
```suggestion
   local Chat = game:GetService("Chat")
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 9]
```diff
@@ -232,18 +232,18 @@ This script used an array to create a list of possible dialogue lines for a Non-
...
 
    -- Add array here
-   local dialogueArray= {"Hi!",  "Do I know you?",  "Goodbye!"}
+   local dialogueArray = {"Hi!",  "Do I know you?",  "Goodbye!"}
    local dialogueIndex = 1
 
    local function speak()
       local dialogue = dialogueArray[dialogueIndex]
-      Chat:Chat(head, dialogue)
+      TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:240]:
```suggestion
      Chat:Chat(head, dialogue)
```

File: content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md [hunk 10]
```diff
@@ -271,6 +271,6 @@ Try one of the optional challenges below.
    local function speak()
       local randomIndex = randomGenerator:NextInteger(1, #dialogueArray)
       local dialogue = dialogueArray[randomIndex]
-      Chat:Chat(head, dialogue)
+      TextChatService:DisplayBubble(head, dialogue)
```
Comment (aetter) [inline content/en-us/tutorials/fundamentals/coding-5/intro-to-arrays.md:274]:
```suggestion
      Chat:Chat(head, dialogue)
```

**Review by aetter — APPROVED** [review]

**Comment by aetter** [comment]
Thanks so much @Flikter for the contribution here and for leading me down a very interesting rabbit hole!

**Pull request merged**